### PR TITLE
Workflow to setup Windows Domain with DC and hosts

### PR DESCRIPTION
**Pull request by zjleblanc**
This change includes a workflow to provide users a small Windows domain with a domain controller and two windows hosts. The added workflow leverages pre-existing job templates from the cloud and windows content:

- Create an AWS Keypair
- Create a VPC
- Create VMs (domain controller, domain computers)
- Create Domain
- Join Domain

windows_domain_workflow

### CODE REVIEW

File: collections/ansible_collections/demo/cloud/roles/aws/tasks/create_vm.yml [hunk 1]
```diff
@@ -36,6 +36,7 @@
     tags:
       blueprint: "{{ aws_blueprint }}"
       purpose: "{{ aws_purpose_tag }}"
+      role: "{{ aws_role_tag | default('demo') }}"
```
Comment (jce-redhat) [inline collections/ansible_collections/demo/cloud/roles/aws/tasks/create_vm.yml:39]: i don't think the demo.cloud.aws role is used anymore when creating VMs, everything was switched over to to the [aws.infrastructure_config_demos](https://github.com/ansible-content-lab/aws.infrastructure_config_demos) collection a while back.

File: collections/ansible_collections/demo/cloud/roles/aws/templates/aws_windows_userdata.j2 [hunk 2]
```diff
@@ -17,7 +17,7 @@ reg delete "HKLM\SOFTWARE\Policies\Microsoft\Windows\WinRM\Service" /v DisableRu
 Set-MpPreference -DisableRealtimeMonitoring $true
 
 # Enable WinRM
-Invoke-WebRequest -Uri https://raw.githubusercontent.com/ansible/ansible/devel/examples/scripts/ConfigureRemotingForAnsible.ps1 -OutFile C:\ConfigureRemotingForAnsible.ps1
+Invoke-WebRequest -Uri https://raw.githubusercontent.com/ansible/ansible-documentation/devel/examples/scripts/ConfigureRemotingForAnsible.ps1 -OutFile C:\ConfigureRemotingForAnsible.ps1
```
Comment (jce-redhat) [inline collections/ansible_collections/demo/cloud/roles/aws/templates/aws_windows_userdata.j2:20]: same as above, i don't think this role is being called so the changes to userdata here won't be used.  unfortunately there doesn't seem to be an easy way to override the userdata template location in the [aws.infrastructure_config_demos.create_vm](https://github.com/ansible-content-lab/aws.infrastructure_config_demos/tree/main/roles/create_vm) role without changing it upstream in that collection.

Comment (jce-redhat) [reply collections/ansible_collections/demo/cloud/roles/aws/templates/aws_windows_userdata.j2:20]: see #78, the ConfigureRemotingForAnsible.ps1 script being pulled in by the aws.infrastructure_config_demos.create_vm role is essentially the same as your updated one here so i don't think this change is needed.

Comment (zjleblanc) [reply collections/ansible_collections/demo/cloud/roles/aws/templates/aws_windows_userdata.j2:20]: @jce-redhat good call on these two, I think I went down this rabbit hole before I realized we were using the external cloud content repo. In fact, I think I do this everytime 😅 I will revert these two changes.

File: collections/ansible_collections/demo/cloud/roles/aws/tasks/create_vm.yml (hunk 1 above)
Comment (zjleblanc) [reply collections/ansible_collections/demo/cloud/roles/aws/tasks/create_vm.yml:39]: removed

File: collections/ansible_collections/demo/cloud/roles/aws/templates/aws_windows_userdata.j2 (hunk 2 above)
Comment (zjleblanc) [reply collections/ansible_collections/demo/cloud/roles/aws/templates/aws_windows_userdata.j2:20]: removing

**Comment by zjleblanc** [comment]
Latest success - running E2E after @jce-redhat consolidation to common
Screenshot 2024-08-22 at 10 31 52 AM

**Review by bbethell-1 — APPROVED** [review]

File: windows/setup.yml [hunk 3]
```diff
@@ -306,3 +352,139 @@ controller_templates:
...
+    survey_enabled: true
+    survey:
+      name: ''
+      description: ''
+      spec:
+        - question_name: AWS Region
+          type: multiplechoice
+          variable: create_vm_aws_region
+          required: true
+          default: us-east-2
```
Comment (jce-redhat) [inline windows/setup.yml:372]: in other surveys i don't think we set a default for AWS region, but i like the idea so i'll put in an issue to do so.

**Review by jce-redhat — COMMENTED** [review]
minor comments, testing the workflow was successful

**Review by jce-redhat — APPROVED** [review]
looks good, we'll just have to remember to remove the known issues section once the intermittent connectivity issue gets resolved.

**Pull request merged**